src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -2813,6 +2822,372 @@ class AlterMaterializedViewStatementSegment(BaseSegment):
     )
 
 
+class CreateFileFormatSegment(BaseSegment):
+    """A snowflake `CREATE FILE FORMAT` statement.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-file-format.html
+    """
+
+    type = "create_file_format_segment"
+    match_grammar = Sequence(
+        "CREATE",
+        Ref("OrReplaceGrammar", optional=True),
+        Sequence("FILE", "FORMAT"),
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("ObjectReferenceSegment"),
+        # TYPE = <FILE_FORMAT> is included in below parameter segments.
+        # It is valid syntax to have TYPE = <FILE_FORMAT> after other parameters.
+        # Below parameters are either Delimited/AnyNumberOf.
+        # Snowflake does allow mixed but this is not supported.
+        # @TODO: Update below when an OptionallyDelimited Class is available.
+        OneOf(
+            Ref("CsvFileFormatTypeParameters"),
+            Ref("JsonFileFormatTypeParameters"),
+            Ref("AvroFileFormatTypeParameters"),
+            Ref("OrcFileFormatTypeParameters"),
+            Ref("ParquetFileFormatTypeParameters"),
+            Ref("XmlFileFormatTypeParameters"),
+        ),
+        Sequence(
+            # Use a Sequence and include an optional CommaSegment here.
+            # This allows a preceding comma when above parameters are delimited.
+            Ref("CommaSegment", optional=True),
+            Ref("CommentEqualsClauseSegment"),
+            optional=True,
+        ),
+    )
+
+
+class CsvFileFormatTypeParameters(BaseSegment):
+    """A Snowflake File Format Type Options segment for CSV.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-file-format.html
+    """
+
+    type = "csv_file_format_type_parameters"
+
+    _file_format_type_parameter = OneOf(
+        Sequence(
+            "TYPE",
+            Ref("EqualsSegment"),
+            RegexParser(
+                r"'?CSV'?",
+                CodeSegment,
+                name="csv_file_type",
+                type="file_type",
+            ),
```

Comment:
Again a `OneOf` would prevent unbalanced quotes. And would allow you to use the quick `StringParser`:

```suggestion
            OneOf(
                StringParser(
                    r"'CSV'",
                    CodeSegment,
                    name="csv_file_type",
                    type="file_type",
                )
                StringParser(
                    r"CSV",
                    CodeSegment,
                    name="csv_file_type",
                    type="file_type",
                )
            ),
```

Same for the rest of the TypeParameters.

Also not convinced we need `name="csv_file_type",`, versus the more generic `name="file_type",`. So we really need a special type for each type? Isn't that obvious from the contents?

Reply:
Should be done!